hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/AbstractSerializationService.java

```diff
@@ -475,7 +530,12 @@ public SerializerAdapter serializerFor(final int typeId) {
         return idMap.get(typeId);
     }
 
-    public SerializerAdapter serializerFor(final Object object) {
+    @SuppressWarnings("checkstyle:npathcomplexity")
+    public SerializerAdapter serializerFor(final Object object, boolean includeSchema) {
+        if (!active) {
```

Comment:
Can you think of any scenario where we set the active to false, but still need to perform some last-second serialization? Maybe during the shutdown of hot-restart enabled cluster?

Reply:
I put this line because of tests in AbstractSerializationServiceTest which tests the behaviour of disposed serializationService. The tests seemed a little bit wrong to me so I have changed the expected exceptions and changed this part as well. 
see `AbstractSerializationServiceTest`